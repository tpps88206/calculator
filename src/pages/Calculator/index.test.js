import React from 'react';

import { render } from '../../utils/test-utils';
import CalculatorPage from './index';

it('renders CalculatorPage without crashing', () => {
  const div = document.createElement('div');
  render(<CalculatorPage />, div);
});

it('renders CalculatorPage with snapshot', () => {
  const { asFragment } = render(<CalculatorPage />);
  expect(asFragment()).toMatchSnapshot();
});
