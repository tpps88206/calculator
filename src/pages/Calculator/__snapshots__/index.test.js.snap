// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CalculatorPage with snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-content-8 vw-100"
  >
    <div
      class="ma-2 text-center"
    >
      <img
        alt="logo"
        src="logo.png"
      />
      <br />
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-containedSizeLarge MuiButton-sizeLarge"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeLarge"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
              />
            </svg>
          </span>
          <p>
            點擊打開計算機
          </p>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiPaper-root MuiCard-root mx-2 my-2 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root"
      >
        <h1
          id="react-calculator"
        >
          React Calculator
        </h1>
        <p>
          <img
            alt="CI/CD"
            src="https://github.com/tpps88206/calculator/workflows/CI/CD/badge.svg"
          />
          <a
            href="https://codecov.io/gh/tpps88206/calculator"
          >
            <img
              alt="Codecov"
              src="https://img.shields.io/codecov/c/github/tpps88206/calculator"
            />
          </a>
          <img
            alt="License"
            src="https://img.shields.io/github/license/tpps88206/calculator"
          />
        </p>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiCard-root mx-2 my-2 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root"
      >
        <h2
          id="demo"
        >
          Github repo
        </h2>
        <p>
          <a
            href="https://github.com/tpps88206/calculator/"
          >
            https://github.com/tpps88206/calculator/
          </a>
        </p>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiCard-root mx-2 my-2 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root"
      >
        <h2
          id="technologies"
        >
          Technologies
        </h2>
        <ul>
          <li>
            Use 
            <strong>
              React Hook
            </strong>
          </li>
          <li>
            Develop state management with 
            <a
              href="https://redux.js.org/"
            >
              Redux
            </a>
             and 
            <a
              href="https://redux-toolkit.js.org/"
            >
              Redux Toolkit
            </a>
            .
          </li>
          <li>
            Develop CSS style with 
            <a
              href="https://material-ui.com/styles/basics/"
            >
              @material-ui/styles
            </a>
          </li>
          <li>
            Develop router management with 
            <a
              href="https://github.com/ReactTraining/react-router"
            >
              React Router
            </a>
            .
          </li>
          <li>
            Write unit test with 
            <strong>
              testing-library
            </strong>
            .
          </li>
          <li>
            Use 
            <a
              href="https://github.com/features/actions"
            >
              Github Actions
            </a>
             to implement CI/CD and deploy on 
            <strong>
              Github Page
            </strong>
            .
          </li>
          <li>
            Use 
            <a
              href="https://material-ui.com/zh/"
            >
              Material UI
            </a>
             to generate the beautiful components with Material Design.
          </li>
          <li>
            Use 
            <strong>
              OOCSS
            </strong>
             to manage the general CSS style, it is following 
            <a
              href="https://getbootstrap.com/"
            >
              Bootstrap
            </a>
            .
          </li>
          <li>
            Use 
            <a
              href="https://github.com/ReactTraining/react-point"
            >
              react-point
            </a>
             to handle touch events.
          </li>
          <li>
            Use 
            <a
              href="https://github.com/STRML/react-draggable"
            >
              react-draggable
            </a>
             to handle Drag and Drop.
          </li>
          <li>
            Use 
            <a
              href="https://lodash.com/"
            >
              Lodash
            </a>
             to control logic methods.
          </li>
          <li>
            Use 
            <a
              href="https://github.com/JedWatson/classnames"
            >
              Classnames
            </a>
             to manage CSS class name.
          </li>
          <li>
            Use 
            <a
              href="https://typicode.github.io/husky/#/"
            >
              Husky
            </a>
             to handle pre-commit event.
          </li>
          <li>
            All icons are using 
            <a
              href="https://material-ui.com/components/material-icons/"
            >
              Material Icon
            </a>
             or 
            <strong>
              SVG
            </strong>
            .
          </li>
          <li>
            Define coding style with 
            <a
              href="https://prettier.io/"
            >
              Prettier
            </a>
             and 
            <strong>
              Eslint
            </strong>
            .
          </li>
        </ul>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiCard-root mx-2 my-2 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root"
      >
        <h2
          id="features"
        >
          Features
        </h2>
        <ul>
          <li>
            Implement the general functions of calculator.
          </li>
          <li>
            In the desktop
            <ul>
              <li>
                The width of calculator will fix in 
                <code>
                  480px
                </code>
                .
              </li>
              <li>
                But when the browser width is smaller than 768px, the width of calculator will fix in 
                <code>
                  360px
                </code>
                .
              </li>
              <li>
                We can drag and drop the calculator to everywhere.
              </li>
            </ul>
          </li>
          <li>
            In the mobile 
            <ul>
              <li>
                Calculator will fix below, and the height is 
                <code>
                  50%
                </code>
                 of the browser.
              </li>
              <li>
                We can not drag and drop but we can scroll it.
              </li>
            </ul>
          </li>
          <li>
            We can open calculator with clicking button and close it with clicking outside.
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
